### DISCUSSION TITLE
Default decision making and acceptance process

### DISCUSSION THREAD

**Discussion by taylor** [lfn-cnti/bestpractices#130] [General]
# Proposal for minimal decision making and acceptance process for most items

The default process covers items which do not need additional discussion and community buy-in. Any item not in the listed in the “exception” list be be accepted after approval from a small number of any community member approvals (see more below).

Items needing further discussion and approval are the exception. They will use a different decision making process and are listed in the Additional Approval Items list below.

Some examples of items using the default minimal number of community approvals include:
- Spelling, grammar, and adding new interested parties
- Adding new interested parties
- New use cases and changes to existing use cases
- Proposing best practices

Below I've broken the updates up based on daily contributions in the [Contributing Guide](CONTRIBUTING.md) and the move involved and adminstrative items listed in our [Governance documents](GOVERNANCE.md)

A pull request which makes the changes as outlined below is at https://github.com/cncf/cnf-wg/pull/131

---

----
GOVERNANCE.md
----

The process outlined in the GOVERNANCE.md delegate the default process to a more relaxed contributing process, outlined in CONTRIBUTING.md, and focus on the exceptions requiring a more thorough decision making process to acheive our desired outcome of fair representation and technical excellence.

Note that the latter decision making process is marked as TBD so that we can focus on the content the community is currently wanting to contributing and discuss (eg. new use cases and best practice proposals).  

# GOVERNANCE.md - Suggested updates

##  Default changes and approval

Changes to content in the CNF WG repository (including adding new content) for any item not listed in the [Additional Approval Items](#Additional-Approval-Items) will follow the acceptance and approval process outlined in the [CONTRIBUTING guide](CONTRIBUTING.md).

## Changes requiring additional approval

Items in the Additional Approval Items list require further discussion and approval before being accepted. The decision making process for items in this list is TBD. 

### Additional Approval Items
1. This list
1. [Charter](https://github.com/cncf/cnf-wg/blob/master/charter.md)
1. [Governance](https://github.com/cncf/cnf-wg/blob/master/governance.md)
1. [Code of Conduct](https://github.com/cncf/cnf-wg/blob/master/code-of-conduct.md)
1. CNF WG "Approved" Best Practices
    - This would be Best Practices which the CNF WG promotes in an approved list. Eg. Baseline Best Practices
    - Proposing a best practice is okay with the default process. Marking it as accepted and listing it as a CNF WG approved Best Practice requires additional review and approval

-----
CONTRIBUTING.md
----

The following are an overview of suggested  updates to https://github.com/cncf/cnf-wg/blob/master/CONTRIBUTING.md#submit-andor-review-pull-requests section

The following changes only require 1 reviewer:
- spelling fixes
- grammar fixes
- adding new interested parties

Changes to other content in the CNF WG repository (including new content) for any item not listed in the [Additional Approval Items](GOVERNANCE.md#Additional-Approval-Items) list require 5 community reviewers to approve.

Any CNF WG community member may be a reviewer. Reviewers may be added by requesting review access (Github READ role works) via a Github issue.

Anyone with merge access may merge the PR after it has the required number of approvals from reviewers.

# CONTRIBUTING.md - Suggested updates in the PR section

## Acceptance
- Spelling, grammar, and new interested parties require 1 reviewer
- Other changes with exception to those listed in the [Governance Additional Approval Items list](GOVERNANCE.md#Additional-Approval-Items) may accepted after approval of 5 community reviewers.

## Steps for creating a new PR:
- Once you have made your change or added new content
- Ensure you are up to date with the `main` branch of the cnf-wg repository
- Open a new [Pull Request](https://github.com/cncf/cnf-wg/pulls)
- Tag at least 5 reviewers

## Steps to accept a PR
1. Reviewers will review the PR and give their feedback: approve, request change, comment w/o approval
1. After required number of approvals from reviewers is reached the PR may be merged

Anyone with merge access can merge the PR after the item has been approved.

**Comment by taylor** [comment]
Speaking more about the benefits of delegating most of this to the community via the CONTRIBUTING process.

Other than 5 items the vast majority of changes can be done quite easily with approval from the community.  The number of approvals for the vast majority can be adjusted as needed as well. Eg. Maybe more things should require 1-2 reviewers.  Maybe the default should be 4 instead of 5.  That's all a community decision for the CONTRIBUTING guide.  Including items like must pass all linting checks if the community wants that in the CONTRIBUTING guide.

**Comment by taylor** [comment]
See PR as well  https://github.com/cncf/cnf-wg/pull/131

**Comment by sishbi** [comment]
I think there should be an initial file for the 'approved-best-practices.md' and I guess a 'draft-best-practices.md' as well.
